fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/data/LoanTermVariationsDataWrapper.java

```diff
@@ -55,26 +55,34 @@ public LoanTermVariationsDataWrapper(final List<LoanTermVariationsData> exceptio
         dueDateIterator = this.dueDateVariation.listIterator();
     }
 
-    public boolean hasVariation(final LocalDate date) {
+    public boolean hasVariation(final LocalDate date, final Boolean isChangeEmiIfRepaymentDateSameAsDisbursementDateEnabled) {
         ListIterator<LoanTermVariationsData> iterator = this.iterator;
-        return hasNext(date, iterator);
+        return hasNext(date, iterator, isChangeEmiIfRepaymentDateSameAsDisbursementDateEnabled);
     }
 
-    private boolean hasNext(final LocalDate date, ListIterator<LoanTermVariationsData> iterator) {
+    private boolean hasNext(final LocalDate date, ListIterator<LoanTermVariationsData> iterator, 
+            final Boolean isChangeEmiIfRepaymentDateSameAsDisbursementDateEnabled) {
         boolean hasVariation = false;
         if (iterator.hasNext()) {
             LoanTermVariationsData loanTermVariationsData = iterator.next();
-            if (!loanTermVariationsData.getTermApplicableFrom().isAfter(date)) {
-                hasVariation = true;
+            if (!isChangeEmiIfRepaymentDateSameAsDisbursementDateEnabled) {
```

Comment:
you can modify the effective date of emi amount  with the configuration.. no need to check all the time